lib/emque/producing/publisher/rabbitmq.rb

```diff
@@ -42,9 +39,15 @@ def publish(topic, message_type, message, key = nil)
               :persistent => true,
               :type => message_type,
               :app_id => Emque::Producing.configuration.app_name,
-              :content_type => "application/json")
+              :content_type => "application/json"
+            )
+
+            if raise_on_failure
+              success = ch.wait_for_confirms
+            else
+              success = true
+            end
 
-            success = ch.wait_for_confirms
             unless success
```

Comment:
Should this now with the `if raise_on_failure`? So that you don't have to artificially set success to true when raise_on_failure is false?